src/git.rs

```diff
@@ -384,17 +336,10 @@ fn build_commits_maps(commits: &Vec<Commit>) -> (CommitsMap, CommitsMap) {
     (parents_map, children_map)
 }
 
-fn to_commit_ref_map(commits: &[Commit]) -> HashMap<&CommitHash, &Commit> {
+fn to_commit_map(commits: &[Commit]) -> CommitMap {
     commits
         .iter()
-        .map(|commit| (&commit.commit_hash, commit))
-        .collect()
-}
-
-fn to_commit_map(commits: Vec<Commit>) -> CommitMap {
-    commits
-        .into_iter()
-        .map(|commit| (commit.commit_hash.clone(), commit))
+        .map(|commit| (commit.commit_hash.clone(), commit.clone()))
```

Comment:
Can we avoid modifying `to_commit_map` by calculating `commit_hashes` before calling `to_commit_map`? (Can we avoid cloning `Commit`?)

Reply:
Done!